### DISCUSSION TITLE
Supporting an onDelete(deletedNodes: Node[], deletedEdges: Edge[])

### DISCUSSION THREAD

**Discussion by daSn0wie** [xyflow/xyflow#3035] [Ideas]
Because the component doesn't support dangling edges, when a node is deleted, a side-effect of deleting attached edges also occurs.  While the component updates properly, reacting to a delete action is difficult b/c the action is broken up into two events that have no knowledge of each other in its reflection in the interface:  onNodesDelete and onEdgesDelete.  

You can see this problem manifest in the SaveRestore example (https://github.com/wbkd/react-flow/tree/main/examples/vite-app/src/examples/SaveRestore).  If you wanted to automatically save/restore the application on events, it would be impossible b/c we have separate handlers for a user delete action of a node.  You would save the edge deletion state, then save the node deletion state as two separate states, when the user action was only one node deletion event.  When you restore, the user would get back the state where the node is there but the edges are gone. Another restore would bring you back to your original state. 

The proposal is to add an onDelete handler which would give you the nodes and the edges that are being deleted by adding the call to the `deleteElements` function here:  https://github.com/wbkd/react-flow/blob/main/packages/core/src/hooks/useReactFlow.ts#L192

**Comment by AsmisAlan** [comment]
any update about this issue ?

**Reply by moklick** [reply]
v12 has a `onDelete` handler (and also an `onBeforeDelete` handler). It's not stable yet, but you can already try it https://github.com/xyflow/xyflow/discussions/3764
